pkg/linstorhelper/client.go

```diff
@@ -18,11 +19,26 @@ import (
 	"github.com/piraeusdatastore/piraeus-operator/v2/pkg/vars"
 )
 
+// optsCache stores the common LINSTOR Client options for all clients. Every reconcile run may create a new client, but
+// they should all share the same Options such as caches and rate limits, provided they point at the same controller.
+var optsCache sync.Map
+
 // Client is a LINSTOR client with convenience functions.
 type Client struct {
 	lapi.Client
 }
 
+// PerClusterOption makes an option that is shared between all instances connecting to the same cluster.
+// This can be used to make a cached shared between different client instances or a rate limiter applied to all
+// clients connected to the same controller.
+func PerClusterOption(opt func() lapi.Option) lapi.Option {
+	return func(c *lapi.Client) error {
+		clusterMap, _ := optsCache.LoadOrStore(c.BaseURL(), &sync.Map{})
+		clusterOpt, _ := clusterMap.(*sync.Map).LoadOrStore(&opt, opt())
```

Comment:
Here we use a pointer to a function as a map key. This seems rather fragile. This could run twice with `opt` being 2 different functions containing identical code and the behavior would be different from when it runs twice with `opt` being the same function.